speaking-portal-project/src/test/file_check.ts

```diff
@@ -64,45 +68,65 @@ export function checkFileReadAccess(file: string) {
      * @param file - The name of the file to check
      */
     fs.access(file, fs.constants.R_OK, (err) => {
-        // TODO: should we separate these checks?
         if (err) throw Error(`FileNotReadable`)
-        else console.log(`${file} is readable`)
-        return true
     })
+    console.log(`${file} is readable`)
+    return true
 }
 
-export function checkWavFile(file: string) {
-    // TODO: Should we be checking the name that is given or look for a .wav file in the directory?
+export async function isWavFile(file: string) {
+    /**
+     * Checks if the specified audio file is .wav
+     *
+     * @remarks
+     * Ensure checkFileExistence() and checkFileReadAccess() are completed prior to this check
+     *
+     * @param file - The name of the file to check
+     */
     if (file.toLowerCase().includes('.wav')) {
-        console.log(`Audio file ${file} has .wav extension.`)
+        console.log(`Audio file ${file} has .wav extension`)
     } else {
-        throw Error(`Supposed audio file ${file} is not a .wav file.`)
+        throw Error(`AudioNotWavFile`)
     }
+    return true
+}
 
-    // TODO: Check if files are valid/not corrupted
-    // should we do this manually or with this module? https://www.npmjs.com/package/wav-file-info
-
-    // TODO: check if audio is 0 seconds long
+export async function isWavFileValid(file: string) {
+    /**
+     * Checks if the specified .wav file is corrupted
+     *
+     * @remarks
+     * Ensure checkFileExistence(), checkFileReadAccess(), and checkWavFile() are completed prior to this check
+     *
+     * @param file - The name of the file to check
+     */
+    wavFileInfo.infoByFilename(file, function(err:any, info:any){
+        if (err) throw Error("CorruptedWavFile");
+        console.log(info);
+      });
+    console.log("Wav file is not corrupted.")
+    return true
 }
 
-export function checkTextFile(file: string) {
-    // TODO: Should we be checking the name that is given or look for a .txt file in the directory?
+export async function isTextFileValid(file: string) {
+    // Check if specified text file is .txt
     if (file.toLowerCase().includes('.txt')) {
-        console.log(`Input text file ${file} has .txt extension.`)
+        console.log(`Input text file ${file} has .txt extension`)
     } else {
         throw Error(`InvalidTextFile`)
     }
 
-    // TODO: should we check if it's corrupted? how?
-
-    // TODO: check if text file is empty
+    // Check if text file is readable in utf8 format and check if empty
     fs.readFile(file, 'utf8', (err, data) => {
-        if (err) throw Error(`Unable to read ${file}`)
+        if (err) throw Error(`UnableToReadTextFile`)
+        console.log(`Text file ${file} is valid`)
         if (data.length > 0) {
```

Comment:
Will we ever need to add a threshold for this value ?

- if the txt file only has 5 chars ? or 10 ?  I don't think it would really matter, but just a thought

Reply:
I think we should stick with whether the text file is empty or not. In Japanese, they use Kanji which can encompass an entire word, so you could write a sentence with 5 characters.